subprojects/build-init/src/main/resources/org/gradle/buildinit/tasks/templates/build.gradle.kts.template

```diff
@@ -0,0 +1,33 @@
+/*
+ * This build file was generated by the Gradle 'init' task.
+ *
+ * This generated file contains a commented-out sample Java project to get you started.
+ * For more details take a look at the Java Quickstart chapter in the Gradle
+ * user guide available at ${ref_userguide_java_tutorial.groovyComment}
+ */
+
+/*
+plugins {
+    // Apply the java plugin to add support for Java
+    java
+}
+
+// In this section you declare where to find the dependencies of your project
+repositories {
+    // Use 'jcenter' for resolving your dependencies.
+    // You can declare any Maven/Ivy/file repository here.
+    jcenter()
+}
+
+// In this section you declare the dependencies for your production and test code
+dependencies {
+    // The production code uses the SLF4J logging API at compile time
+    compile("org.slf4j:slf4j-api:${slf4jVersion.groovyString}")
+
+    // Declare the dependency for your favourite test framework you want to use in your tests.
+    // TestNG is also supported by the Gradle Test task. Just change the
+    // testCompile dependency to testCompile("org.testng:testng:6.8.1") and add
+    // 'val test by tasks.getting(Test::class) { useTestNG() }' to your build script.
+    testCompile("junit:junit:${junitVersion.groovyString}")
```

Comment:
Why are you choosing to use the `:` based version in these templates instead of the named parameter version.
`testCompile(group = "junit", name = "junit", version = junitVersion.groovyString)`

Personally, I'm a fan of the of the named parameter version. The first thing that I did when I ran the generator for groovy a few months ago was convert all of the dependency lines to the other syntax.

I might just be [bikeshedding](http://bikeshed.com/) here.

Reply:
The generated build scripts, using the Groovy DSL and now the Kotlin DSL, use the `g:n:v` string notation. I didn't want to change this for Groovy and didn't want to have the generated Kotlin DSL to be too different. Moreover, you can find this notation used widely on the interwebs for copy/pasting (plugin portal, search.maven.org, jcenter etc...) so I would say it's not a bad idea to use it when generating builds. Bikeshedding indeed ;)

Reply:
Thanks for the feedback and thoughts on this. Your analysis makes sense.